File: app/src/main/java/com/example/myfirstkotlinapp/Sprites/MoleHole.java
package com.example.myfirstkotlinapp.Sprites;

import android.graphics.Bitmap;
import android.graphics.Canvas;

public class MoleHole extends BaseSprite {

    private Bitmap backgroundTexture;
    private Bitmap foregroundTexture;

    public MoleHole(Vector2 position, Bitmap backgroundTexture, Bitmap foregroundTexture) {
        super(position);

        this.backgroundTexture = backgroundTexture;
        this.foregroundTexture = foregroundTexture;
    }

    @Override
    public void draw(Canvas canvas) {
        canvas.drawBitmap(this.backgroundTexture, this.position.x * this.backgroundTexture.getWidth(), this.position.y * this.backgroundTexture.getHeight(), null);
    }

    public void drawForeground(Canvas canvas) {
        canvas.drawBitmap(this.foregroundTexture, this.position.x * this.foregroundTexture.getWidth(), this.position.y * this.foregroundTexture.getHeight(), null);
    }
}

File: app/src/main/java/com/example/myfirstkotlinapp/sprites/MoleHole.kt
package com.example.myfirstkotlinapp.sprites

import android.graphics.Bitmap
import android.graphics.Canvas
import com.example.myfirstkotlinapp.gamecore.BaseSprite
import com.example.myfirstkotlinapp.gamecore.Vector2

class MoleHole(
    position: Vector2,
    private val backgroundTexture: Bitmap,
    private val foregroundTexture: Bitmap) : BaseSprite(position) {

    override fun draw(canvas: Canvas?) {
        canvas?.drawBitmap(
            backgroundTexture,
            position.x * backgroundTexture.width,
            position.y *backgroundTexture.height,
            null
        )
    }

    fun drawForeground(canvas: Canvas?) {
        canvas?.drawBitmap(
            foregroundTexture,
            position.x * foregroundTexture.width,
            position.y *foregroundTexture.height,
            null
        )
    }

    override fun update() {
        // Nothing needed here yet.
    }
}